doc/examples/functional_usage.rst

```diff
@@ -0,0 +1,39 @@
+Using Renate Functionally
```

Comment:
IMO the title is too generic and not very accessible.

Maybe we should think about something like "Using Renate components to train a model in a custom script" ?